### PR TITLE
consensus: increase log level for corner case when prevoting a proposal

### PR DESCRIPTION
**Pull request by cason**
Addresses #1309.

---

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

File: internal/consensus/state.go [hunk 1]
```diff
@@ -1457,8 +1457,8 @@ func (cs *State) defaultDoPrevote(height int64, round int32) {
 	*/
 	blockID, ok := cs.Votes.Prevotes(cs.Proposal.POLRound).TwoThirdsMajority()
 	ok = ok && !blockID.IsNil()
-	if ok && cs.ProposalBlock.HashesTo(blockID.Hash) && cs.Proposal.POLRound >= 0 && cs.Proposal.POLRound < cs.Round {
```
Comment (cason) [inline internal/consensus/state.go:1460]: POLRound > -1 here, by construction. Moreover, the `Proposal.ValidateBasic()` rejects POLRound < -1.

**Comment by angbrav** [comment]
LGTM, thanks for taking care of it @cason! The logging is definitely in place.

I wonder if in such a worrying scenario: there is evidence that two quorum certificates have been formed for conflicting proposals, one should rather panic as malicious parties may have been able to violate safety and fork the chain. But this is probably a separate discussion.

**Comment by cason** [comment]
> one should rather panic as malicious parties may have been able to break safety and fork the chain.

We have other situations where this may happen. This occurs when the assumption 2/3+ majority of honest processes is violated. We do not panic in this case, but I do agree that we should log this to make debugging easier.

**Review by melekes — APPROVED** [review]
👍

**Comment by cason** [comment]
I was considering whether to backport this change. But I realized that the code of consensus `cs.defaultDoPrevote` method is substantially different (was refactored) on the release branches. So I would not.
